### PR TITLE
Selection sort

### PR DESCRIPTION
**Pull request by finico**
see title :)

### CODE REVIEW

File: selection_sort/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            smallest = arr[i]
+
+    return smallest_index
+
+
+def selection_sort(arr):
+    res = []
+
+    for _ in range(len(arr)):
+        smallest = find_smallest(arr)
```
Comment (codewylle) [inline selection_sort/__init__.py]: `smallest` in this function is not consistent with `smallest` in `find_smallest`, man will be confused. `smallest_index` is better

File: selection_sort/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+def find_smallest(arr):
+    smallest_index = 0
+    smallest = arr[smallest_index]
```
Comment (codewylle) [inline selection_sort/__init__.py]: You can omit `smallest` and use only `smallest_index`

Comment (finico) [reply selection_sort/__init__.py]: `smallest` was necessary for comparison with value of `arr[i]`

File: selection_sort/__init__.py (hunk 1 above)
Comment (finico) [reply selection_sort/__init__.py]: Man moved the function into the bag

File: selection_sort/__init__.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    for i in range(len(res) - 1):
+        least = i
+
+        for j in range(least + 1, len(res)):
+            if res[j] < res[least]:
+                least = j
+
+        temp = res[i]
+        res[i] = res[least]
+        res[least] = temp
```
Comment (codewylle) [inline selection_sort/__init__.py]: 1 line instead of 3: `res[i], res[least] = res[least], res[i]`

File: selection_sort/__init__.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+def selection_sort(arr):
+    res = arr.copy()
+
+    for i in range(len(res) - 1):
+        least = i
+
+        for j in range(least + 1, len(res)):
+            if res[j] < res[least]:
+                least = j
+
```
Comment (codewylle) [inline selection_sort/__init__.py]: hm, it's hard to read. `least = find_smallest(res, i)`

File: selection_sort/test_selection_sort.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+from big_o import big_o, complexities, datagen
+
+from selection_sort import selection_sort
+
+
+class TestSelectionSort(TestCase):
+    def test_sort(self):
+        self.assertEqual(selection_sort([2, 3, 1, 5, 4]), [1, 2, 3, 4, 5])
+
```
Comment (codewylle) [inline selection_sort/test_selection_sort.py:11]: Check source array is not mutated

File: selection_sort/__init__.py (hunk 4 above)
Comment (finico) [reply selection_sort/__init__.py]: Yup, splitted

File: selection_sort/__init__.py (hunk 3 above)
Comment (finico) [reply selection_sort/__init__.py]: Sure Man, I forgot about this feature of Python

File: selection_sort/test_selection_sort.py (hunk 5 above)
Comment (finico) [reply selection_sort/test_selection_sort.py:11]: Test was added, Man

**Review by codewylle — APPROVED** [review]
Good job, man

**Pull request merged**